src/Cake.Common/Net/HttpAliases.cs

```diff
@@ -197,7 +197,7 @@ public static void DownloadFile(this ICakeContext context, Uri address, FilePath
             // We track the last posted value since the event seems to fire many times for the same value.
             var percentComplete = 0;
 
-            using (var http = new HttpClient())
+            using (var http = new HttpClient(new HttpClientHandler { UseDefaultCredentials = settings.UseDefaultCredentials }))
```

Comment:
It might not `make any sense` but right now, based on your current implementation, it is possible, and without testing out the code, I am trying to get to the bottom of what `would` happen.  I would have thought that something like this makes sense:

```
using (var http = settings.UseDefaultCredentials ? new HttpClient(new HttpClientHandler { UseDefaultCredentials = true }): new HttpClient())
```

following by, within the settings:

```
if (!settings.UseDefaultCredentials && !string.IsNullOrWhiteSpace(settings.Username) && !string.IsNullOrWhiteSpace(settings.Password))
```

That way, you are forcing the user to either use one authentication mechanism, or the other, but not both at the same time.